realm/src/androidTest/java/io/realm/RealmInMemoryTest.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm;
+
+import android.os.StrictMode;
+import android.test.AndroidTestCase;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import io.realm.entities.Dog;
+
+public class RealmInMemoryTest extends AndroidTestCase {
+
+    private final static String IDENTIFIER = "InMemRealmTest";
+
+    private Realm testRealm;
+    private RealmConfiguration inMemConf;
+
+    @Override
+    protected void setUp() throws Exception {
+        RealmConfiguration onDiskConf = new RealmConfiguration.Builder(getContext())
+                .name(IDENTIFIER)
+                .build();
+        inMemConf = new RealmConfiguration.Builder(getContext())
+                .name(IDENTIFIER)
+                .inMemory()
+                .build();
+
+        // Delete the same name Realm file just in case
+        Realm.deleteRealm(onDiskConf);
+        testRealm = Realm.getInstance(inMemConf);
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        if (testRealm != null) {
+            testRealm.close();
+        }
+    }
+
+    // Testing the in-memory Realm by Creating one instance, adding a record, then close the instance.
+    // By the next time in-memory Realm instance with the same name created, it should be empty.
+    // Use StrictMode to check no disk IO would happen in VM to this thread.
+    public void testInMemoryRealm() {
+        StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder()
```

Comment:
On IO here:
```
        public Builder(Context context) {
            if (context == null) {
                throw new IllegalArgumentException("A non-null Context must be provided");
            }
            initializeBuilder(context.getFilesDir());
        }
```
I am thinking to keep it to prevent from doing something wrong in the future?